rs/https_outcalls/client/src/client.rs

```diff
@@ -264,15 +270,16 @@ impl NonBlockingChannel<CanisterHttpRequest> for CanisterHttpAdapterClientImpl {
                 canister_id: request_sender,
                 content: match adapter_canister_http_response.await {
                     Ok(resp) => {
-                        metrics.request_total.with_label_values(&["success"]).inc();
+                        metrics.request_total.with_label_values(&["success", http_method_label]).inc();
                         CanisterHttpResponseContent::Success(resp)
                     },
                     Err((reject_code, message)) => {
-                        metrics.request_total.with_label_values(&[&reject_code.to_string()]).inc();
+                        metrics.request_total.with_label_values(&[&reject_code.to_string(), http_method_label]).inc();
```

Comment:
Does this work?
```suggestion
                        metrics.request_total.with_label_values(&[reject_code.to_str(), http_method_label]).inc();
```
My understanding is that we are currently coverting `RejectCode` to `&str`, then `&str` to `String`, and then take a reference to `String` (https://sourcegraph.com/github.com/dfinity/ic/-/blob/packages/ic-error-types/src/lib.rs?L32), which seems unnecessary